src/librustc/hir/intravisit.rs

```diff
@@ -382,10 +382,10 @@ pub trait Visitor<'v>: Sized {
 }
 
 /// Walks the contents of a crate. See also `Crate::visit_all_items`.
-pub fn walk_crate<'v, V: Visitor<'v>>(visitor: &mut V, krate: &'v Crate) {
+pub fn walk_crate<'v, V: Visitor<'v>>(visitor: &mut V, krate: &'v Crate<'v>) {
     visitor.visit_mod(&krate.module, krate.span, CRATE_HIR_ID);
-    walk_list!(visitor, visit_attribute, &krate.attrs);
-    walk_list!(visitor, visit_macro_def, &krate.exported_macros);
+    walk_list!(visitor, visit_attribute, krate.attrs);
```

Comment:
I'm curious, why didn't `&krate.attrs` still work?

Reply:
I changed the `HirVec<T>` to be `&'hir [T]`. Using `&krate.attrs` used to make a `&[T]`, and now makes a `&&[T]`.